### PR TITLE
Variable material for porous flow NS (PINSFV)

### PR DESCRIPTION
**Pull request by GiudGiud**
We could also:
- modify the INSFV one to provide speed (not really a big deal, but would force another rename)
- Keep providing rho_cp_T through the enthalpy material and provide the other missing functors through another material

Let me know what you think

Name could also be PINSFVVarMaterial to make it simpler. I am not sure there will ever be a PINSFV with conserved variables.

### CODE REVIEW

**Comment by lindsayad** [comment]
> I am not sure there will ever be a PINSFV with conserved variables

No I don't think so either. To me the 'I' in INSFV and PINSFV is equal to Rhie-Chow which in implementation basically always uses primitive pressure and velocity variables

**Comment by lindsayad** [comment]
> * Keep providing rho_cp_T through the enthalpy material and provide the other missing functors through another material

I think this would be my preference. An INSFVSpeedMaterial or something. With the action I feel like the cons of an additional material are lessened

**Comment by GiudGiud** [comment]
Works for me. I updated it to this.

At this point we could just add a pre-factor to `VectorMagnitudeFunctorMaterial`, would just be missing the NS namespace terminology and the check. But that could be done in the action..

**Review by lindsayad — CHANGES_REQUESTED** [review]
Can we rename the class from INSFV to PINSFV? I'm sorry if my previous comment led you down the path of calling it INSFV

File: modules/navier_stokes/doc/content/source/materials/INSFVSpeedFunctorMaterial.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# PINSFVSpeedFunctorMaterial
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/materials/INSFVSpeedFunctorMaterial.md:1]: Need to `git mv` this

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/doc/content/source/materials/INSFVSpeedFunctorMaterial.md (hunk 1 above)
Comment (GiudGiud) [reply modules/navier_stokes/doc/content/source/materials/INSFVSpeedFunctorMaterial.md]: that and the tests design doc lol
should be good now

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
@grmnptr this is in, let's add it to the action

**Pull request merged**